src/main/java/com/mraof/minestuck/client/gui/playerStats/EcheladderScreen.java

```diff
@@ -190,8 +196,46 @@ else if(textColors.length > rung)
 				fill(poseStack, xOffset + 90, y + 10, xOffset + 90 + (int)(146* ClientPlayerData.getRungProgress()), y + 12, bg);
 			} else rand.nextInt(0xFFFFFF);
 			
-			String s = I18n.exists("echeladder.rung."+rung) ? I18n.get("echeladder.rung."+rung) : "Rung "+(rung+1);
-			mc.font.draw(poseStack, s, xOffset+ladderXOffset - mc.font.width(s) / 2, y + 2, textColor);
+			
+			String s = null;
+			if(playerTitle != null)
+			{
+				//this is all the clauses for the echeladder
+				String playerBoots = String.valueOf((Minecraft.getInstance().player).getInventory().armor.get(0));
+				String playerLeggings = String.valueOf((Minecraft.getInstance().player).getInventory().armor.get(1));
+				String playerChestplate = String.valueOf((Minecraft.getInstance().player).getInventory().armor.get(2));
+				String playerHat = String.valueOf((Minecraft.getInstance().player).getInventory().armor.get(3));
+				String playerClass = (playerTitle.getHeroClass().toString());
+				String playerAspect = (playerTitle.getHeroAspect().toString());
+				String playerColor = String.valueOf(ClientPlayerData.getPlayerColor());
+				String playerBoon = String.valueOf(ClientPlayerData.getBoondollars());
+				String playerModus = String.valueOf(ClientPlayerData.getModus());
+				String consortRep = String.valueOf(ClientPlayerData.getConsortReputation());
+				String wearingArmorSet = playerChestplate + playerBoots + playerHat + playerLeggings;
+				
+				if(I18n.exists("echeladder.rung." + rung + "." + playerAspect + "." + playerClass))
+				{
+					s = I18n.get("echeladder.rung." + rung + "." + playerAspect + "." + playerClass);
+				} else if(I18n.exists("echeladder.rung." + rung + "." + playerColor))
+				{
+					s = I18n.get("echeladder.rung." + rung + "." + playerColor);
+				} else if(I18n.exists("echeladder.rung." + rung + "." + playerAspect))
+				{
+					s = I18n.get("echeladder.rung." + rung + "." + playerAspect);
+				}
+			}
+			if(s == null)
+			{
+				if(I18n.exists("echeladder.rung." + rung))
+				{
+					s = I18n.get("echeladder.rung." + rung);
+				} else
+				{
+					s = "Rung " + (rung + 1);
+				}
+			}
```

Comment:
As was mentioned in en_us.json, these changes are not necessarily appropriate to what the PR intends to accomplish. Not all of these clauses are represented in en_us, and due also to that lack of completion, this would be better suited for a separate PR